src/sentry/api/serializers/models/alert_rule_trigger.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import absolute_import
+
+from collections import defaultdict
+
+import six
+
+from sentry.api.serializers import register, Serializer
+from sentry.incidents.models import AlertRuleTrigger, AlertRuleTriggerExclusion
+
+
+@register(AlertRuleTrigger)
+class AlertRuleTriggerSerializer(Serializer):
+    def serialize(self, obj, attrs, user):
+        return {
+            "id": six.text_type(obj.id),
+            "alertRuleId": six.text_type(obj.alert_rule_id),
+            "label": obj.label,
+            "thresholdType": obj.threshold_type,
+            "alertThreshold": obj.alert_threshold,
+            "resolveThreshold": obj.resolve_threshold,
+            "dateAdded": obj.date_added,
+        }
+
+
+class DetailedAlertRuleTriggerSerializer(AlertRuleTriggerSerializer):
+    def get_attrs(self, item_list, user, **kwargs):
+        triggers = {item.id: item for item in item_list}
+        result = defaultdict(dict)
+        for trigger_id, project_slug in AlertRuleTriggerExclusion.objects.filter(
+            alert_rule_trigger__in=item_list
+        ).values_list("alert_rule_trigger_id", "query_subscription__project__slug"):
```

Comment:
Should this be `query_subscription__project_slug`? I thought `__` indicated association traversal in django.

Reply:
It does, but you need to use it to access variables too. Otherwise if the relationship was called `project_thing` then with`query_subscription__project_thing_variable` we wouldn't know where the table name ended and the variable began. The mental model is to just replace `__` with `.`.